From: Arvind Yadav <arvind.yadav.cs@gmail.com>
Date: Sun, 16 Jul 2017 12:00:44 +0530
Subject: infiniband: mthca: constify pci_device_id.
Patch-mainline: v4.14-rc1
Git-commit: e0604df111e57773f45c50d143fedb0374bfe916
References: bsc#1046306 FATE#322942

pci_device_id are not supposed to change at runtime. All functions
working with pci_device_id provided by <linux/pci.h> work with
const pci_device_id. So mark the non-const structs as const.

File size before:
   text	   data	    bss	    dec	    hex	filename
  13067	    805	      4	  13876	   3634	infiniband/hw/mthca/mthca_main.o

File size After adding 'const':
   text	   data	    bss	    dec	    hex	filename
  13419	    453	      4	  13876	   3634	infiniband/hw/mthca/mthca_main.o

Signed-off-by: Arvind Yadav <arvind.yadav.cs@gmail.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/mthca/mthca_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/mthca/mthca_main.c
+++ b/drivers/infiniband/hw/mthca/mthca_main.c
@@ -1161,7 +1161,7 @@ static void mthca_remove_one(struct pci_
 	mutex_unlock(&mthca_device_mutex);
 }
 
-static struct pci_device_id mthca_pci_table[] = {
+static const struct pci_device_id mthca_pci_table[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_MELLANOX, PCI_DEVICE_ID_MELLANOX_TAVOR),
 	  .driver_data = TAVOR },
 	{ PCI_DEVICE(PCI_VENDOR_ID_TOPSPIN, PCI_DEVICE_ID_MELLANOX_TAVOR),
